### PR TITLE
Setup typescript

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,24 +2,29 @@
...
     "build": "cheminfo-build --entry src/index.js --root SpectraFitting",
+    "check-types": "tsc --noEmit",
+    "clean": "rimraf lib lib-esm",
     "eslint": "eslint src",
     "eslint-fix": "npm run eslint -- --fix",
-    "compile": "rollup -c",
-    "prepublishOnly": "npm run compile",
-    "test": "npm run test-coverage && npm run eslint",
+    "prepack": "npm run tsc",
+    "test": "npm run test-coverage && npm run eslint && npm run check-types",
```
Comment (targos) [inline package.json:20]:
```suggestion
    "test": "npm run test-only && npm run eslint && npm run check-types",
```

File: package.json [hunk 2]
```diff
@@ -2,24 +2,29 @@
...
     "prettier": "prettier --check src",
     "prettier-write": "prettier --write src",
     "test-coverage": "jest --coverage",
     "test-only": "jest",
```
Comment (targos) [inline package.json:24]:
```suggestion
    "test-only": "jest --coverage",
```

File: package.json [hunk 3]
```diff
@@ -41,20 +46,21 @@
...
-    "eslint": "^8.5.0",
+    "eslint": "^8.6.0",
     "eslint-config-cheminfo": "^7.2.1",
+    "eslint-config-cheminfo-typescript": "^10.3.0",
     "esm": "^3.2.25",
     "jest": "^27.4.5",
     "prettier": "^2.5.1",
-    "rollup": "^2.61.1",
-    "spectrum-generator": "^6.0.2"
+    "rollup": "^2.62.0",
```
Comment (targos) [inline package.json:60]:
```suggestion
```

File: src/spectra-fitting.d.ts [hunk 4]
```diff
@@ -1,16 +1,17 @@
 import { DataXY } from 'cheminfo-types';
```
Comment (targos) [inline src/spectra-fitting.d.ts:1]: The file name should be `spectra-fitting.ts`

**Comment by wadjih-bencheikh18** [comment]
@lpatiny could you please check if there any other problems

**Pull request merged**